spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -542,6 +542,52 @@ case class AlterTableChangeColumnDeltaCommand(
   }
 }
 
+/**
+ * A command to replace columns for a Delta table, support changing the comment of a column,
+ * reordering columns, and loosening nullabilities.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   ALTER TABLE table_identifier REPLACE COLUMNS (col_spec[, col_spec ...]);
+ * }}}
+ */
+case class AlterTableReplaceColumnsDeltaCommand(
+    table: DeltaTableV2,
+    columns: Seq[StructField])
+  extends LeafRunnableCommand with AlterDeltaTableCommand with IgnoreCachedData {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val deltaLog = table.deltaLog
+    recordDeltaOperation(deltaLog, "delta.ddl.alter.replaceColumns") {
+      val txn = startTransaction(sparkSession)
+
+      val metadata = txn.metadata
+      val existingSchema = metadata.schema
+
+      val resolver = sparkSession.sessionState.conf.resolver
+      val changingSchema = StructType(columns)
+
+      SchemaUtils.canChangeDataType(existingSchema, changingSchema, resolver,
+        txn.metadata.columnMappingMode, failOnAmbiguousChanges = true).foreach { operation =>
+        throw DeltaErrors.alterTableReplaceColumnsException(
+          existingSchema, changingSchema, operation)
+      }
```

Comment:
aside: That method's name is the opposite of what it actually does... shouldn't it really have a name like `SchemaUtils.checkDataTypeChange`?

Reply:
That would be a great follow up PR, I don't want to make such a large change at the moment (it may cause renames around the codebase)